jnigen/test/simple_package_test/java/com/github/dart_lang/jnigen/generics/MyMap.java

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+package com.github.dart_lang.jnigen.generics;
+
+import java.lang.reflect.Array;
+import java.util.HashMap;
+import java.util.Map;
+
+public class MyMap<K, V> {
+  public class MyEntry {
```

Comment:
Nested classes inherit type arguments? Funky.

What happens in Java if you instantiate two different MyMap's with different type arguments and try to pass add one entry to another? Is there a static check? (At runtime there's just erasure...)

For us in Dart we must unnest the classes so we have the type arguments in the entry class as well.

Reply:
There is a static check. MyMap<A, B>.MyEntry won't match MyMap<C, D>.MyEntry.